lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -332,19 +323,55 @@ def run_batch(
         :param as_admin: Allows you to run jobs in other people's workspaces
         :return: A list of condor job ids or a failure notification
         """
-        wsid = batch_params.get("wsid")
+        if type(params) != list:
+            raise IncorrectParamsException("params must be a list")
+        wsid = batch_params.get(_WORKSPACE_ID)
         meta = batch_params.get("meta")
         if as_admin:
             self.sdkmr.check_as_admin(requested_perm=JobPermissions.WRITE)
         else:
             # Make sure you aren't running a job in someone elses workspace
             self._check_workspace_permissions(wsid)
-            wsids = [job_input.get("wsid", wsid) for job_input in params]
+            # this is very odd. Why check the parent wsid again if there's no wsid in the job?
+            # also, what if the parent wsid is None?
+            # also also, why not just put all the wsids in one list and make one ws call?
+            wsids = [job_input.get(_WORKSPACE_ID, wsid) for job_input in params]
             self._check_workspace_permissions_list(wsids)
 
+        self._add_job_requirements(params)
+        self._check_job_arguments(params)
+
         parent_job = self._create_parent_job(wsid=wsid, meta=meta)
         children_jobs = self._run_batch(parent_job=parent_job, params=params)
-        return {"parent_job_id": str(parent_job.id), "child_job_ids": children_jobs}
+        return {_PARENT_JOB_ID: str(parent_job.id), "child_job_ids": children_jobs}
+
+    # modifies the jobs in place
+    def _add_job_requirements(self, jobs):
+        # could add a cache in the job requirements resolver to avoid making the same
+        # catalog call over and over if all the jobs have the same method
+        jrr = self.sdkmr.get_job_requirements_resolver()
+        for j in jobs:
+            # TODO JRR check if requesting any job requirements & if is admin
+            # TODO JRR actually process the requirements once added to the spec
+            j[_JOB_REQUIREMENTS] = jrr.resolve_requirements(j.get(_METHOD))
```

Comment:
This is where the future run-time job requirements feature will be added - user provided requirements will be provided to the resolver much like the concierge params below